src/booking/booking.service.ts

```diff
@@ -56,6 +60,114 @@ export class BookingService {
     }
   }
 
+  async bookAccommodation(userId: string, body: BookingReqDto) {
+    const { accommodationId, startDate, endDate } = body;
+
+    try {
+      const bookingStart = dayjs(startDate);
+      const bookingEnd = dayjs(endDate);
+
+      if (!bookingStart.isBefore(bookingEnd, 'day') || !bookingStart.isAfter(dayjs().endOf('day')))
+        throw new BadRequestException('Invalid booking dates');
+
+      const accommodation = await this.prismaService.accommodation.findUnique({
+        where: { id: accommodationId },
+        select: {
+          availableFrom: true,
+          availableTo: true,
+          booking: {
+            select: {
+              startDate: true,
+              endDate: true,
+            },
+            where: {
```

Comment:
looks like not better approach. you don't know why db doesn't contain such record. because of id or condition. better pull it by id and check right here if startdate and enddate are fit